lib/Models/TableStyle.js

```diff
@@ -37,163 +46,53 @@ var TableStyle = function(options) {
-     * Gets or sets a string or {@link ColorMap} array, specifying how to map values to colors.  Setting this property sets
-     * {@link TableStyle#colorPalette} to undefined.  If this property is a string, it specifies a list of CSS colors separated by hyphens (-),
-     * and the colors are evenly spaced over the range of values.  For example, "red-white-hsl(240,50%,50%)".
-     * @memberOf TableStyle.prototype
-     * @type {String}
-     * @see TableStyle#colorPalette
-     */
-    if (defined(options.colorMap)) {
-        this.colorMap = new ColorMap(options.colorMap);
-    } else {
-        this.colorMap = undefined;
-    }
+};
 
-    /**
-     * Gets or sets the [ColorBrewer](http://colorbrewer2.org/) palette to use when mapping values to colors.  Setting this
-     * property sets {@link TableStyle#colorMap} to undefined.  This property is ignored if {@link TableStyle#colorMap} is defined.
-     * @memberOf TableStyle.prototype
-     * @type {String}
-     * @see  TableStyle#colorMap
-     */
-    this.colorPalette = options.colorPalette;  // Only need this here so that updateFromJson sees colorPalette as a property.
-    if (defined(this.colorPalette)) {
-        var that = this;
-        return ColorMap.loadFromPalette(this.colorPalette).then(function(colorMap) {
-            that.colorMap = colorMap;
-        });
-    }
+inherit(TableColumnStyle, TableStyle);
 
-    /**
-     * How many horizontal ticks to draw on the generated color ramp legend, not counting the top or bottom.
-     * @type {Integer}
-     */
-    this.legendTicks = defaultValue(options.legendTicks, 3);
+// When columns is updated via json, turn it into TableColumnStyle objects.
+// Do this in updateFromJson so we can keep track of the promises required by the colorPalette option.
+// This also has the advantage of not using the TableColumnStyle constructor to set the properties, which causes problems with colorPalette too.
+TableStyle.prototype.updateFromJson = function(json, options) {
+    var promises = [updateFromJson(this, json, options)];
+    this.columns = objectToTableColumnStyle(json, promises, options);
+    return when.all(promises);
 };
 
-// When colorMap is updated, we need to convert it to a colorMap.
-// When colorPalette is updated, we need to update colorMap.
-TableStyle.prototype.updaters = {
-    colorMap: function(tableStyle, json, propertyName) {
-        tableStyle.colorMap = new ColorMap(json[propertyName]);
-    },
-    colorPalette: function(tableStyle, json, propertyName) {
-        return ColorMap.loadFromPalette(json[propertyName]).then(function(colorMap) {
-            tableStyle.colorMap = colorMap;
-        });
-    }
+TableStyle.prototype.updaters = clone(TableStyle.prototype.updaters);
+TableStyle.prototype.updaters['columns'] = function(tableStyle, json, propertyName) {
+    // Do not update columns here; do it directly in the updateFromJson function.
+    // tableStyle.columns = objectToTableColumnStyle(json[propertyName]);
```

Comment:
My idea is, by including an empty function here, lib/Core/updateFromJson.js calls this (and does nothing) instead of doing `target[propertyName] = json[propertyName];` , ie. it does not set `columns`.  
Instead, columns is set in the preceding function, the tableStyle's own `updateFromJson`.
I did it this way because TableColumnStyle's colorPalette actually returns a promise. The updaters can't handle a promise, but updateFromJson can.
Of course, I settled on this admittedly strange pattern after all the other things I tried failed...